bin/jruby.bash

```diff
@@ -125,41 +115,40 @@ add_log "  JAVA_OPTS: $JAVA_OPTS"
 
 # ----- Discover JVM and prep environment to run it ---------------------------
 
-# Detect cygwin, darwin, and mingw environments
-case "`uname`" in
+# Detect cygwin and mingw environments
+case "$(uname)" in
   CYGWIN*) cygwin=true;;
-  Darwin) darwin=true;;
   MINGW*) jruby.exe "$@"; exit $?;;
 esac
 
 # Determine where the java command is and ensure we have a good JAVA_HOME
 if [ -z "$JAVACMD" ] ; then
   if [ -z "$JAVA_HOME" ] ; then
     JAVACMD='java'
-    JAVA_HOME=$(dirname $(dirname `which java`))
+    JAVA_HOME="$(dirname "$(dirname "$(command -v java)")")"
   else
     if $cygwin; then
-      JAVACMD="`cygpath -u "$JAVA_HOME"`/bin/java"
+      JAVACMD="$(cygpath -u "$JAVA_HOME")/bin/java"
     else
       JAVACMD="$JAVA_HOME/bin/java"
     fi
   fi
 else
-  expanded_javacmd=`which $JAVACMD`
-  if [[ -z "$JAVA_HOME" && -x $expanded_javacmd ]] ; then
-    JAVA_HOME=$(dirname $(dirname $expanded_javacmd))
+  expanded_javacmd="$(command -v "$JAVACMD")"
+  if [ -z "$JAVA_HOME" ] && [ -x "$expanded_javacmd" ] ; then
+    JAVA_HOME="$(dirname "$(dirname "$expanded_javacmd")")"
   fi
 fi
 
 # Detect Java 9+ by the presence of a jmods directory in JAVA_HOME
-if [ -d $JAVA_HOME/jmods ]; then
+if [ -d "$JAVA_HOME"/jmods ]; then
```

Comment:
Note: this actually fixes a warning running on Windows in a Bash shell:
```
/c/Users/runneradmin/.rubies/jruby-9.2.9.0/bin/jruby: line 155: [: C:\Program: binary operator expected
```
This happens in GitHub Actions.

Reply:
Makes sense with spaces in path.

Reply:
Yeah, exactly. Using the Bash launcher Windows is likely unusual though (Just `jruby` would give precedence to the `.exe`, only `/full/path/to/jruby` or `jruby.bash` in a Bash shell would use the Bash launcher).